### ISSUE TITLE
Partial and Unlabeled Predicitions 

### ISSUE THREAD

**Issue by nicholasprimiano** [longzw1997/Open-GroundingDino#25]
After fine tuning on on a small set of classes. I get a bunch of predictions are are unlabeled or only contain part of the label name. For example, I have a class called ''mvr" and it is displayed as 'mv'. I have attached an image of the output below.

![image](https://github.com/longzw1997/Open-GroundingDino/assets/16170448/8b350a2b-0ff5-4391-a447-3a1ecc85aec1)

**Comment by BIGBALLON** [comment]
I'm not sure of the exact reason.  I think you can refer [Explanations/Tips for Grounding DINO Inputs and Outputs](https://github.com/IDEA-Research/GroundingDINO#star-explanationstips-for-grounding-dino-inputs-and-outputs) in official repo or test the label one by one to check the problem. BTW, the detection of medical images may not be easy to train, and false positives may be reasonable, we may be able to find the reason from multiple aspects or even data enhancement during training.

**Comment by BIGBALLON** [comment]
I'm not sure of the exact reason.  I think you can refer [Explanations/Tips for Grounding DINO Inputs and Outputs](https://github.com/IDEA-Research/GroundingDINO#star-explanationstips-for-grounding-dino-inputs-and-outputs) in official repo or test the label one by one to check the problem. 
BTW, the detection of medical images may not be easy to train, I'm not sure these terms(avr, mvr, defib etc.) are reasonable for LLM, false positives may be reasonable. Maybe fine-tuning the OVD model is not the best way, it would be better to train a purely visual object detection model?? We may be able to find the reason from multiple aspects or even data enhancement during training.

**Comment by nicholasprimiano** [comment]
The finetuned model produces interesting output. MVR = mitral value  repair. On this image form the test dataset it correctly identifies the mitral valve repair but only if I prompt it with "mitral valve repai". I will not work if I use "mitral valve repair" as text input. If I use "mvr" (the term I used for fine tuning), the model again correctly identifies the object but call it "mv". I wonder if there is an issue related to the tokenizer.

![image](https://github.com/longzw1997/Open-GroundingDino/assets/16170448/5148564e-da69-4a94-9afe-33017b19708b)
![image](https://github.com/longzw1997/Open-GroundingDino/assets/16170448/7a700048-a88c-40e8-84ab-a185cf52f6dd)
![image](https://github.com/longzw1997/Open-GroundingDino/assets/16170448/d1a896ec-4e72-41d7-8c30-5e24a41d185d)

**Comment by BIGBALLON** [comment]
from [Explanations/Tips for Grounding DINO Inputs and Outputs](https://github.com/IDEA-Research/GroundingDINO#star-explanationstips-for-grounding-dino-inputs-and-outputs)

Note that each word can be split to more than one tokens with different tokenlizers. The number of words in a sentence may not equal to the number of text tokens.

so tokenizer is the key


maybe OWLv2 or GLIP is better
